pkg/apiserver/handlers/featuregates/handler.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package featuregates
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"gopkg.in/yaml.v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/controller/querier"
+	"antrea.io/antrea/pkg/features"
+	"antrea.io/antrea/pkg/util/env"
+)
+
+var controllerOnlyGates = sets.NewString("Traceflow", "AntreaPolicy", "Egress", "NetworkPolicyStats")
+
+type (
+	Config struct {
+		// FeatureGates is a map of feature names to bools that enable or disable experimental features.
+		FeatureGates map[string]bool `yaml:"featureGates,omitempty"`
+	}
+
+	Response struct {
+		Component      string `json:"component,omitempty"`
+		FeatureName    string `json:"name,omitempty"`
+		FeatureStatus  string `json:"status,omitempty"`
+		FeatureVersion string `json:"version,omitempty"`
+	}
+)
+
+const (
+	controllerMode = "controller"
+	agentMode      = "agent"
+)
+
+// HandleFunc returns the function which can handle queries issued by 'antctl get featuregates' command.
+// The handler function populates Antrea featuregates information to the response.
+// For now, it will return all feature gates included in features.DefaultAntreaFeatureGates for agent
+// We need to exclude any new feature gates which is not consumed by agent in the future.
+func HandleFunc(cq querier.ControllerQuerier) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		k8sClientset := cq.GetK8sClient()
```

Comment:
Can you pass k8sClient directly to `HandleFunc` instead of injecting it to `ControllerQuerier` and then pass in it? 
Not this PR, it seems the `ControllerQuerier` has been misused and becoming a container of dependencies. But I would avoid adding more unnecessary dependencies to it.

Reply:
@tnqn I added a new feild in extraConfig to pass k8s client to controller apiserver. and all other comments have been addressed as well, please help to review when you have a time, thanks.